storm-core/test/jvm/org/apache/storm/messaging/netty/NettyTest.java

```diff
@@ -0,0 +1,382 @@
+                            waitUntilReady(client, server.get());
+                        } catch (Exception e) {
+                            throw Utils.wrapInRuntime(e);
+                        }
+                    });
+                    serverStart.get(Testing.TEST_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+                    byte[] messageBytes = reqMessage.getBytes(StandardCharsets.UTF_8);
+
+                    client.send(taskId, messageBytes);
+
+                    waitForNotNull(response);
+                    TaskMessage responseMessage = response.get();
+                    assertThat(responseMessage.task(), is(taskId));
+                    assertThat(responseMessage.message(), is(messageBytes));
+                } finally {
+                    if (server.get() != null) {
+                        server.get().close();
+                    }
+                }
+            }
+        } finally {
+            context.term();
+        }
+    }
+
+    @Test
+    public void testServerDelayed() throws Exception {
+        doTestServerDelayed(basicConf());
+    }
+
+    @Test
+    public void testServerDelayedWithSasl() throws Exception {
+        doTestServerDelayed(withSaslConf(basicConf()));
+    }
+
+    private void doTestBatch(Map<String, Object> stormConf) throws Exception {
+        int numMessages = 100000;
+        LOG.info("Should send and receive many messages (testing with " + numMessages + " messages)");
+        ArrayList<TaskMessage> responses = new ArrayList<>();
+        AtomicInteger received = new AtomicInteger();
+        IContext context = TransportFactory.makeContext(stormConf);
+        try {
+            try (IConnection server = context.bind(null, 0);
+                IConnection client = context.connect(null, "localhost", server.getPort(), remoteBpStatus)) {
+                server.registerRecv(mkConnectionCallback((message) -> {
+                    responses.add(message);
+                    received.incrementAndGet();
+                }));
+                waitUntilReady(client, server);
+
+                IntStream.range(1, numMessages)
+                    .forEach(i -> client.send(taskId, String.valueOf(i).getBytes(StandardCharsets.UTF_8)));
+
+                Testing.whileTimeout(Testing.TEST_TIMEOUT_MS,
+                    () -> responses.size() < numMessages - 1,
+                    () -> {
+                        LOG.info("{} of {} received", responses.size(), numMessages - 1);
+                        sleep().run();
+                    });
+                IntStream.range(1, numMessages)
+                    .forEach(i -> {
+                        assertThat(new String(responses.get(i - 1).message(), StandardCharsets.UTF_8), is(String.valueOf(i)));
+                    });
+            }
+        } finally {
+            context.term();
+        }
+    }
+
+    @Test
+    public void testBatch() throws Exception {
+        doTestBatch(basicConf());
```

Comment:
This conf should have `STORM-MESSAGING-NETTY-BUFFER-SIZE 1024000` as `test-batch` defines, so you may want to define another method for new configuration.